src/modules/staking/services/staking.filtering.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Inject, Injectable, forwardRef } from '@nestjs/common';
+import { StakingFarmsFilter } from '../models/staking.args';
+import { StakingService } from './staking.service';
+
+@Injectable()
+export class StakingFilteringService {
+    constructor(
+        @Inject(forwardRef(() => StakingService))
+        private readonly stakingService: StakingService,
+    ) {}
+
+    async stakingFarmsByToken(
+        stakingFilter: StakingFarmsFilter,
+        stakeAddresses: string[],
+    ): Promise<string[]> {
+        if (
+            !stakingFilter.searchToken ||
+            stakingFilter.searchToken.trim().length < 3
+        ) {
+            return stakeAddresses;
+        }
+
+        const searchTerm = stakingFilter.searchToken.toUpperCase().trim();
+
+        const farmTokens = await Promise.all(
```

Comment:
I would keep only the `farmingToken` for this filtering

Reply:
done